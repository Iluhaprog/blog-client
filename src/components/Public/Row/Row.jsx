import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const RowBox = styled.div`
  display: flex;
  flex-direction: row;
  justify-content: ${({justifyContent}) => justifyContent};
  align-items: ${({alignItems}) => alignItems};
`;

const Row = ({justifyContent, alignItems, children}) => {
  return (
    <RowBox justifyContent={justifyContent} alignItems={alignItems}>
      {children}
    </RowBox>
  );
};

Row.defaultProps = {
  justifyContent: 'flex-start',
};

Row.propTypes = {
  justifyContent: PropTypes.oneOf([
    'flex-start',
    'flex-end',
    'center',
    'space-between',
    'space-around',
  ]),
  alignItems: PropTypes.oneOf([
    'flex-start',
    'flex-end',
    'start',
    'end',
    'center',
    'stretch',
  ]),
  children: PropTypes.node,
};

export {Row};
