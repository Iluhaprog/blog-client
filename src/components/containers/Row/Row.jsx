import React from 'react';

export default ({ children, justifyContent, alignItems, wrap }) => {
    const jc = justifyContent ? `row_jc-${justifyContent}` : '';
    const ai = alignItems ? `row_ai-${alignItems}` : ''
    const w = wrap ? `row_w-${wrap}` : ''; 
    return (
        <div className={`row ${jc} ${ai} ${w}`}>
            { children }
        </div>
    );
}