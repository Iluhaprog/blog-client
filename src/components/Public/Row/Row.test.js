import {render} from '@testing-library/react';
import {Row} from './Row';

test('Should render Header component', () => {
  const {asFragment} = render(
    <Row justifyContent={'center'} alignItems={'start'}>
      Test header
    </Row>
  );
  expect(asFragment()).toMatchSnapshot();
});
