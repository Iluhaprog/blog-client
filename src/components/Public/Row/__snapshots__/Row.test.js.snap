// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render Header component 1`] = `
<DocumentFragment>
  <div
    class="sc-bdfBwQ iHiDst"
  >
    Test header
  </div>
</DocumentFragment>
`;
